code/game/objects/effects/mines.dm

```diff
@@ -51,6 +59,7 @@
 	name = "shrapnel mine"
 	var/shrapnel_type = /obj/projectile/bullet/shrapnel
 	var/shrapnel_magnitude = 3
+	del_self = FALSE

```

Comment:
I don't understand why is this necessary. Please explain the issue and solution.

Reply:
A pellet cloud is a component that is used to generate shrapnel, and is added to things like grenades and landmines. When the grenade or landmine goes off, the cloud component shoots a bunch of shrapnel everywhere. To cut down on spam for someone getting hit by multiple pellets, the pellet clouds track each pellet and what they hit, so when all the pellets have hit something or run out of range, we tell the person they're hit by 5 pellets instead of 5 separate messages.

Because tracking the pellets requires the component staying around after the grenade or landmine goes off, we nullspace the mine or grenade rather than qdel'ing it outright to allow the component to survive, then qdel the grenade/mine and component once the pellets are finished.

If you'd like I can add more documentation for my reasoning

Reply:
This should be handled by the component then, hook up to pre delete signal, prevent destruction. move to nullspace then delete after component is done.